src/topics/inputs.md

```diff
@@ -143,13 +143,85 @@ The `baseCommand` field will always appear in the final command line before the
 
 [touch]: http://www.linfo.org/touch.html
 
+### Optional Input Parameters
+
+Optional `inputs` include `label` and `secondaryFiles`.
+`label` is a short, human-readable description for the parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    label: Use case for label parameter
+    inputBinding:
+      position: 1
+```
+
+`secondaryFiles` is an optional input parameter that provides a pattern of specifying files
+or directories that must be included alongside the primary file. 
+The following example demonstrates the `secondaryFiles` input parameter'
+
+```cwl
+inputs:
+  example_file:
+    type: file
+    inputBinding:
+      position: 1
+      prefix: --file=
+      seperate: false
+    "secondaryFiles": [
+            {
+                "class": "File",
+                "location": "example.txt",
+                "basename": "example.txt",
+                "nameroot": "example",
+                "nameext": ".txt",
+            }
+        ]
+```
+If the value is an expression, the value of self in the expression must be the primary
+input or output File object to which this binding applies.
+The `basename`, `nameroot` and `nameext` fields must be present in self.
+
+Also, a file object listed in secondaryFiles may contain nested secondaryFiles as shown below:
+
+```cwl
```

Comment:
We should use a `code-block` here, which supports extra arguments to allow for linking, line-number highlighting, etc. See other code-blocks with CWL files. I think the contributing file should include something about these too.